Virtual Keyboard/Assets/Scripts/CategoryManager.cs

```diff
@@ -247,4 +247,24 @@ public class CategoryManager : MonoBehaviour {
 		}
 	}
 
+	[ContextMenu("Add New Category")]
+	void addCategory() {
+		Debug.Log ("Add New Category");
+
+		GameObject newCategory = new GameObject ();
+		newCategory.transform.SetParent (transform);
+		newCategory.name = "New Category";
+		newCategory.tag = "Category";
+		newCategory.transform.localScale = Vector3.one;
+		newCategory.transform.localPosition = Vector3.zero;
+		newCategory.transform.localRotation = Quaternion.Euler (Vector3.zero); 
+
+		ListManager listManager = newCategory.AddComponent<ListManager> ();
+		listManager.title = "New Category";
```

Comment:
In order to get the number you can init the categories list and call findCategories as in the comment below then just look at categegories.count